index.js

```diff
@@ -11,11 +11,21 @@ function toObject(val) {
 	return Object(val);
 }
 
-function isFunc(val) {
-	return typeof val === 'function';
+function assignKey(to, from, key) {
+	var val = from[key];
+
+	if (typeof val === 'undefined' || val === null) {
+		return;
+	}
+
+	if (typeof to[key] === 'undefined' || !isObj(val)) {
```

Comment:
What do you think? Is it ok to clone the object if the target is `null`?
The original `Object.assign` throws an error, but it could be annoying in the recursive fashion.

Wouldn't it be better if it moved the source object by reference? I'm just asking.

```js
if (typeof to[key] === 'undefined' || to[key] === null || !isObj(val)) { ...
```
test:
```js
test('move plain object by reference if target is null', t => {
	var bar = {};
	var target = {foo: null};
	var source = {foo: bar};
	fn(target, source);
	t.is(target.foo, bar);
	t.end();
});
```